SYCL/ESIMD/api/functional/mutator.hpp

```diff
@@ -0,0 +1,87 @@
+//===-- mutator.hpp - This file provides common function and classes to mutate
+//      reference data. ---------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides common function and classes to mutate reference data.
+///
+//===----------------------------------------------------------------------===//
+
+#include "value.hpp"
+#include <sycl/sycl.hpp>
+
+#include <algorithm>
+#include <vector>
+
+#pragma once
+
+namespace esimd_test::api::functional {
+
+// Provides base functions to mutate value.
+namespace mutator {
+
+template <typename T> class Add {
+  T m_value;
+
+public:
+  Add(T value_for_mutation) : m_value(value_for_mutation) {}
+
+  void operator()(T &value_for_mutation) {
```

Comment:
`value_for_mutation` has been renamed into `val` [here](https://github.com/intel/llvm-test-suite/pull/789/commits/7cc2f674c94df0416edeeb133af0de198829cfb6#diff-fa18495d278b576095ad26f3fbce0799c62e2302a8837d4e0a289c922d0dce74R60) and in all cases
I can't use `value` name because we have another one `value` that uses for obtain lowest, max, etc. values